File: 0401_0500/0466/0466.py
def func(n):
    while not (n & 1):
        n >>= 1
    return 1 if n == 1 else func(n // 2) + func(n // 2 + 1)

with open("input.txt", "r") as f, open("output.txt", "w") as q:
    q.write(str(func(int(f.readline()))))

File: 0401_0500/0466/0466.c
#include <stdio.h>

int	func(int n)
{
	while (!(n & 1))
		n >>= 1;
	return n == 1 ? 1 : func(n / 2) + func(n / 2 + 1);
}

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int n;

	fscanf(f, "%d", &n);
	fprintf(q, "%d", func(n));
	return 0;
}
